langchainlaw/generate_prompts.py

```diff
@@ -1,91 +1,119 @@
-    prompts = prompts.fillna('')
-    prompt_responses = defaultdict(lambda: {'fields': defaultdict(dict), 'return_type': '', 'repeats': '', 'prompt_question': '', 'return_instruction': '', 'additional_instruction': ''})
-    # for each row in the prompts sheet, extract the prompt and the response
-
-    for i, row in prompts.iterrows():
-        if row['return_type']:
-            prompt_responses[row['Prompt_name']]['return_type'] = row['return_type']
-        if row['repeats']:
-            prompt_responses[row['Prompt_name']]['repeats'] = row['repeats']
-        if row['prompt_question']:
-            prompt_responses[row['Prompt_name']]['prompt_question'] = row['prompt_question']
-        if row['return_instruction']:
-            prompt_responses[row['Prompt_name']]['return_instruction'] = row['return_instruction']
-        if row['additional_instruction']:
-            prompt_responses[row['Prompt_name']]['additional_instruction'] = row['additional_instruction']
-        
-        prompt_responses[row['Prompt_name']]['fields'][row['fields']]['question_description'] = row['question_description']
-        prompt_responses[row['Prompt_name']]['fields'][row['fields']]['example'] = row['example']
-        prompt_responses[row['Prompt_name']]['fields'][row['fields']]['question_number'] = len(prompt_responses[row['Prompt_name']]['fields'])
+    prompts = pd.read_excel(input_file, sheet_name="sample_prompts", dtype=str).fillna(
+        ""
+    )
+
+    prompt_responses = defaultdict(
+        lambda: {
+            "fields": defaultdict(dict),
+            "return_type": "",
+            "repeats": "",
+            "prompt_question": "",
+            "return_instruction": "",
+            "additional_instruction": "",
+        }
+    )
+
+    for _, row in prompts.iterrows():
+        prompt = prompt_responses[row["Prompt_name"]]
+
+        if row["return_type"]:
+            prompt.update(
+                {
+                    "return_type": row["return_type"],
+                    "repeats": row["repeats"],
+                    "prompt_question": row["prompt_question"],
+                    "return_instruction": row["return_instruction"],
+                    "additional_instruction": row["additional_instruction"],
+                }
+            )
+
+        field = prompt["fields"][row["fields"]]
+        field.update(
+            {
+                "question_description": row["question_description"],
+                "example": row["example"],
+                "question_number": len(prompt["fields"]),
+            }
+        )
 
     return prompt_responses
 
 
 def generate_random_page_reference():
-    return [f"(p{random.randint(1, 100)})", f"(pp{random.randint(1, 5)}-{random.randint(5, 10)})"][random.randint(0, 1)]
+    return [
+        f"(p{random.randint(1, 100)})",
+        f"(pp{random.randint(1, 5)}-{random.randint(6, 10)})",
+    ][random.randint(0, 1)]
+
 
 def assemble_prompts_to_yaml(prompts: dict):
```

Comment:
I think you should try yaml.dump(object) - this should return a YAML string from the data stucture, rather than building it up as strings - just to see if the output works

Reply:
yaml.dump(object) returns a yaml string but it will be tricky to get it to match the format of the expected prompts...?

Reply:
Yeah - let's just leave it as it is for now - I'll add an issue